packages/google_maps_flutter/ios/Classes/GoogleMapPolylineController.m

```diff
@@ -0,0 +1,178 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import "GoogleMapPolylineController.h"
+#import "JsonConversions.h"
+
+@implementation FLTGoogleMapPolylineController {
+  GMSPolyline* _polyline;
+  GMSMapView* _mapView;
+}
+- (instancetype)initPolylineWithPath:(GMSMutablePath*)path
+                          polylineId:(NSString*)polylineId
+                             mapView:(GMSMapView*)mapView {
+  self = [super init];
+  if (self) {
+    GMSMutablePath* path = [GMSMutablePath path];
+    _polyline = [GMSPolyline polylineWithPath:path];
+    _mapView = mapView;
+    _polylineId = polylineId;
+    _polyline.userData = @[ polylineId ];
+  }
+  return self;
+}
+
+- (void)removePolyline {
+  _polyline.map = nil;
+}
+
+#pragma mark - FLTGoogleMapPolylineOptionsSink methods
+
+- (void)setConsumeTapEvents:(BOOL)consumes {
+  _polyline.tappable = consumes;
+}
+- (void)setVisible:(BOOL)visible {
+  _polyline.map = visible ? _mapView : nil;
+}
+- (void)setZIndex:(int)zIndex {
+  _polyline.zIndex = zIndex;
+}
+- (void)setPoints:(NSMutableArray*)points {
+  GMSMutablePath* path = [GMSMutablePath path];
+
+  for (NSObject* point in points) {
+    CLLocation* location = (CLLocation*)point;
+    [path addCoordinate:location.coordinate];
+  }
+  _polyline.path = path;
+}
+
+- (void)setColor:(UIColor*)color {
+  _polyline.strokeColor = color;
+}
+- (void)setStrokeWidth:(CGFloat)width {
+  _polyline.strokeWidth = width;
+}
+@end
+
+static int ToInt(NSNumber* data) { return [FLTGoogleMapJsonConversions toInt:data]; }
+
+static BOOL ToBool(NSNumber* data) { return [FLTGoogleMapJsonConversions toBool:data]; }
+
+static NSMutableArray* ToPoints(NSArray* data) {
+  return [FLTGoogleMapJsonConversions toPoints:data];
+}
+
+static UIColor* ToColor(NSArray* data) { return [FLTGoogleMapJsonConversions toColor:data]; }
+
+static void InterpretPolylineOptions(NSDictionary* data, id<FLTGoogleMapPolylineOptionsSink> sink,
+                                     NSObject<FlutterPluginRegistrar>* registrar) {
+  NSNumber* consumeTapEvents = data[@"consumeTapEvents"];
+  if (consumeTapEvents) {
+    [sink setConsumeTapEvents:ToBool(consumeTapEvents)];
+  }
+
+  NSNumber* visible = data[@"visible"];
+  if (visible) {
+    [sink setVisible:ToBool(visible)];
+  }
+
+  id zIndex = data[@"zIndex"];
```

Comment:
Can `zIndex`, `points`, `strokeColor`, `strokeWidth` have different types? If their types are known, declare them as specific type instead of `id`

Reply:
Done.